src/streams.h

```diff
@@ -684,6 +684,26 @@ class CBufferedFile
         }
     }
 
+    //! move the read position ahead in the stream to the given position
+    void SkipTo(const uint64_t file_pos) {
+        assert(file_pos >= m_read_pos);
+        // Use SetPos() to back up in the stream, not SkipTo().
+        if (file_pos > nReadLimit) {
+            throw std::ios_base::failure("Attempt to position past buffer limit");
+        }
+        while (m_read_pos < file_pos) {
+            if (m_read_pos == nSrcPos)
+                Fill();
```

Comment:
Good observation, and actually the same applies to the `read()` method. I looked into making this optimization, but I think it would make this PR too complex; maybe it can be done in a follow-up, but I don't think it's important in practice.